### PR TITLE
Plugin documentation added

### PR DESCRIPTION
**Pull request by rg663**
Resolves issue #472 — Documentation to link a Django app, or plugin, to ColdFront is added to the ColdFront documentation at https://coldfront.readthedocs.io/en/latest/.

### CODE REVIEW

File: docs/pages/plugin/how_to_create_a_plugin.md [hunk 1]
```diff
@@ -0,0 +1,110 @@
+# How to Create Your Own Plugin For ColdFront
+
+ColdFront plugins are synonymous to Django apps. To create a plugin, all you have to do is create a Django app and link it to ColdFront by editing configuration settings.
+
+The tutorial below assumes that you are comfortable with Django. If not, check out [these docs]("https://docs.djangoproject.com/en/4.2/intro/tutorial01/") for an introduction to how to build a Django app.
```
Comment (dsajdak) [inline docs/pages/plugin/how_to_create_a_plugin.md:5]: This link doesn't work.  I think you need to take out the quotes around the URL

File: docs/pages/plugin/how_to_create_a_plugin.md [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+    name = 'coldfront.core.weeklyreportapp'
+```
+
+4. Make sure to include this line in your **urls.py** file!
+```
+app_name = "weeklyreportapp"
+```
+## Link Your App to ColdFront
+
+1. Add the following app to your list of ```INSTALLED_APPS``` in your ColdFront **base.py** file:
```
Comment (dsajdak) [inline docs/pages/plugin/how_to_create_a_plugin.md:48]:
```suggestion
1. Add the following app to your list of ```INSTALLED_APPS``` in the ColdFront **base.py** (`coldfront/config/base.py`) file:
```

File: docs/pages/plugin/how_to_create_a_plugin.md [hunk 3]
```diff
@@ -0,0 +1,110 @@
...
+```
+# ColdFront Apps
+INSTALLED_APPS += [
+    'coldfront.core.weeklyreportapp',
+]
+```
+2. Edit your main ColdFront **urls.py** file to include the new urls:
```
Comment (dsajdak) [inline docs/pages/plugin/how_to_create_a_plugin.md:55]:
```suggestion
2. Edit the ColdFront **urls.py** (`coldfront/config/urls.py`) file to include the new urls info:
```

File: docs/pages/plugin/how_to_create_a_plugin.md [hunk 4]
```diff
@@ -0,0 +1,110 @@
...
+│   models.py
+│   tests.py
+│   urls.py
+│   views.py
+│
+└───templates (you can include as many as needed)
+    │   index.html
+
+```
+2. Ensure that your **__init__.py** file contents have the following format:
```
Comment (dsajdak) [inline docs/pages/plugin/how_to_create_a_plugin.md:27]:
```suggestion
2. Ensure that your app's **__init__.py** file contents have the following format:
```

File: docs/pages/plugin/how_to_create_a_plugin.md [hunk 5]
```diff
@@ -0,0 +1,110 @@
...
+```
+default_app_config = "coldfront.core.weeklyreportapp.apps.WeeklyreportappConfig"
+```
+
+3. Ensure that your **apps.py** file contents look like this:
```
Comment (dsajdak) [inline docs/pages/plugin/how_to_create_a_plugin.md:32]:
```suggestion
3. Ensure that your app's **apps.py** file contents look like this:
```

File: docs/pages/plugin/how_to_create_a_plugin.md [hunk 6]
```diff
@@ -0,0 +1,110 @@
...
+```
+from django.apps import AppConfig
+
+
+class WeeklyreportappConfig(AppConfig):
+    default_auto_field = 'django.db.models.BigAutoField'
+    name = 'coldfront.core.weeklyreportapp'
+```
+
+4. Make sure to include this line in your **urls.py** file!
```
Comment (dsajdak) [inline docs/pages/plugin/how_to_create_a_plugin.md:42]:
```suggestion
4. Make sure to include this line in your app's **urls.py** file:
```

**Review by dsajdak — CHANGES_REQUESTED** [review]
@rg663 this all works in my testing but I did recommend some changes in the docs.  I'd also like to talk to you about how we can better recommend connecting plugins to CF.  It works to put this in the core directory but I think it might be better to recommend putting them in the plugins directory - or provide examples for both.  You may have had technical reasons to do this or it ties into your work on #537

File: docs/pages/plugin/how_to_create_a_plugin.md [hunk 7]
```diff
@@ -0,0 +1,116 @@
...
+```
+{% extends "common/base.html" %}
+{% load crispy_forms_tags %}
+{% load humanize %}
+{% load static %}
+```
+
+## Link Your App to ColdFront
+
+1. Add the following app to your list of ```INSTALLED_APPS``` by creating a new file in the ColdFront plugins directory (`coldfront/plugins`) and adding these lines to the new file:
```
Comment (dsajdak) [inline docs/pages/plugin/how_to_create_a_plugin.md:70]:
```suggestion
1. Add the following app to your list of ```INSTALLED_APPS``` by creating a new file in the ColdFront plugins directory (`coldfront/congig/plugins`) and adding these lines to the new file (i.e. `weeklyreportapp.py`):
```

File: .gitignore [hunk 8]
```diff
@@ -21,3 +21,45 @@ db.json
 .env
 .devcontainer/*
 .bin/*
+coldfront/core/allocation/migrations/0006_auto_20230607_1719.py
```
Comment (aebruno) [inline .gitignore:24]: Not sure why all these are getting added to `.gitignore`. Lets remove these.

**Review by aebruno — CHANGES_REQUESTED** [review]
just need to fix up .gitignore. otherwise looks good

**Review by aebruno — APPROVED** [review]

**Pull request merged**